services/blockchain-indexer/shared/indexer/tokenIndex.js

```diff
@@ -18,39 +18,50 @@ const {
 	MySQL: { getTableInstance },
 } = require('lisk-service-framework');
 
+const dataService = require('../dataService');
 const { requestConnector } = require('../utils/request');
 
+const { isMainchain } = require('../chain');
+
 const config = require('../../config');
 
 const topLSKAddressesIndexSchema = require('../database/schema/topLSKAddresses');
 
 const MYSQL_ENDPOINT = config.endpoints.mysql;
 
+let liskTokenID;
+const LOCAL_ID_LSK = '00000000';
+
 const getTopLSKAddressesIndex = () => getTableInstance(
 	topLSKAddressesIndexSchema.tableName,
 	topLSKAddressesIndexSchema,
 	MYSQL_ENDPOINT,
 );
 
 const getLiskBalanceByAddress = async (address) => {
-	const LISK_TOKEN_ID = config.tokens.lisk.id;
+	if (!liskTokenID) {
+		const { chainID } = (await dataService.getNetworkStatus()).data;
+		liskTokenID = chainID.concat(LOCAL_ID_LSK);
+	}
 
 	const response = await requestConnector(
 		'getTokenBalance',
 		{
 			address,
-			tokenID: LISK_TOKEN_ID,
+			tokenID: liskTokenID,
 		},
 	);
 
 	return response.availableBalance;
 };
 
 const updateLiskBalance = async (job) => {
-	const { address } = job.data;
-	const balance = await getLiskBalanceByAddress(address);
-	const topLSKAddressesDB = await getTopLSKAddressesIndex();
-	await topLSKAddressesDB.upsert({ address, balance });
+	if (await isMainchain()) {
+		const { address } = job.data;
+		const balance = await getLiskBalanceByAddress(address);
+		const topLSKAddressesDB = await getTopLSKAddressesIndex();
+		await topLSKAddressesDB.upsert({ address, balance });
+	}
```

Comment:
Let us currently index this information on all the chains.
```suggestion
	const { address } = job.data;
	const balance = await getLiskBalanceByAddress(address);
	const topLSKAddressesDB = await getTopLSKAddressesIndex();
	await topLSKAddressesDB.upsert({ address, balance });
```